LiteCore/Database/DatabaseImpl.cc

```diff
@@ -576,12 +576,29 @@ namespace litecore {
                 DebugAssert(isValidScopeNameOrDefault(scope));
                 name.setStart(slash + 1);
             }
+            // CBL-3824: This is necessary for determining if the keystore is an index, which may occur in
+            // the case of a Full-Text Index. The keystore name for an FTI looks like: <scope>.<collection>::<index>
+            slice idxName = nullslice;
+            // Find the first ':'
+            if (auto idxSep = name.findByte(KeyStore::kIndexSeparator)) {
+                idxName = slice(idxSep+1, name.end());
+                // Find the second colon and set index name accordingly
+                if (auto idxSep2 = idxName.findByte(KeyStore::kIndexSeparator); idxSep2 == idxSep+1) {
+                    idxName.setStart(idxSep2+1);
+                    // make sure name does not contain the index name, otherwise later assertion will fail
+                    name.setEnd(idxSep);
+                } else {
+                    idxName = nullslice;
+                }
+            }
+
             DebugAssert((name == kC4DefaultCollectionName && scope == kC4DefaultScopeID)
                         || KeyStore::isValidCollectionName(name));
-            return {name, scope};
-        } else {
-            return {nullslice, nullslice};
+            // If keystore name was not an index name
+            if(!idxName)
```

Comment:
It is okay, but in general empty() is a better predicate, for size == 0 makes !!idName == true

Reply:
I simply did that because if the first or second colon are not found, idxName is set to nullslice. Would .empty() be invalid on a nullslice?

Reply:
It does not matter in this case, because you are so close to where it is set.